bin/author.rb

```diff
@@ -1,7 +1,11 @@
 class Author
   attr_accessor :first_name, :last_name
 
-  def initialize(first_name, last_name)
+  def self.from_hash(data)
+    new(data['first_name'], data['last_name'], data['id'])
+  end
+
+  def initialize(first_name, last_name, id = nil)
     @id = Random.rand(1...1000)
```

Comment:
Please, remember to assign your new id parameter if it's given to the constructor:
Something like this: `@id = id || Random.rand(1...1000)`

Also, you must not assign @id in this constructor, you should call super-class constructor using super() method, take a look at the Book file.